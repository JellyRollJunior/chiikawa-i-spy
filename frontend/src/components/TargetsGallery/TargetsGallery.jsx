import styles from './TargetsGallery.module.css';
import shared from '../../styles/shared.module.css';
import usagi from '../../assets/temp/usagi.jpg';
import shisa from '../../assets/temp/shisa.jpg';
import hachiware from '../../assets/temp/hachiware.jpg';
import chiikawa from '../../assets/temp/chiikawa.jpg';

const tempImages = [usagi, shisa, hachiware, chiikawa];

const TargetsGallery = ({ targets }) => {
  return (
    <section className={styles.targetSection}>
      <ul className={shared.horizontalWrapper}>
        {targets.map((target, index) => (
          <li className={styles.target} key={target.id}>
            <img
              src={tempImages[index]}
              alt={target.name}
              className={`${styles.targetImage} ${
                target.isFound && styles.foundImage
              }`}
            />
            <h4 className={target.isFound ? `${styles.foundTitle}` : ''}>
              {target.name}
              {target.isFound && ' âœ“'}
            </h4>
          </li>
        ))}
      </ul>
    </section>
  );
};

export { TargetsGallery };
