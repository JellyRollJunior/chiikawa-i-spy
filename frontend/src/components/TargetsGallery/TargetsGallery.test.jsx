import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { TargetsGallery } from './TargetsGallery.jsx';

const TARGETS_DATA = [
  {
    id: 3,
    name: 'Floppy-Eared Usagi',
    isFound: false,
  },
  {
    id: 4,
    name: 'Satapanbin Shisa',
    isFound: false,
  },
  {
    id: 5,
    name: 'Guitar Hachiware',
    isFound: false,
  },
  {
    id: 6,
    name: 'Aura Chiikawa',
    isFound: false,
  },
];

describe('Targets Gallery component', () => {
  it('renders targets with images and names', () => {
    const { container } = render(<TargetsGallery targets={TARGETS_DATA} />);

    expect(container).toMatchSnapshot();
  });

  it('renders found targets with a checkmark', () => {
    const name = 'Floppy-Eared Usagi';
    const target = [
      {
        id: 3,
        name,
        isFound: true,
      },
    ];
    render(<TargetsGallery targets={target} />);

    expect(screen.getByRole('heading', { name: `${name} âœ“`})).toBeInTheDocument();
  });
});
