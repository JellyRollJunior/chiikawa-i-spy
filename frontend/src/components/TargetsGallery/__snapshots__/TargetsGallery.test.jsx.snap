// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Targets Gallery component > renders targets with images and names 1`] = `
<div>
  <section
    class="_targetSection_c5ad1c"
  >
    <ul
      class="_horizontalWrapper_5890fc"
    >
      <li
        class="_target_c5ad1c"
      >
        <img
          alt="Floppy-Eared Usagi"
          class="_targetImage_c5ad1c false"
          src="/src/assets/temp/usagi.jpg"
        />
        <h4
          class=""
        >
          Floppy-Eared Usagi
        </h4>
      </li>
      <li
        class="_target_c5ad1c"
      >
        <img
          alt="Satapanbin Shisa"
          class="_targetImage_c5ad1c false"
          src="/src/assets/temp/shisa.jpg"
        />
        <h4
          class=""
        >
          Satapanbin Shisa
        </h4>
      </li>
      <li
        class="_target_c5ad1c"
      >
        <img
          alt="Guitar Hachiware"
          class="_targetImage_c5ad1c false"
          src="/src/assets/temp/hachiware.jpg"
        />
        <h4
          class=""
        >
          Guitar Hachiware
        </h4>
      </li>
      <li
        class="_target_c5ad1c"
      >
        <img
          alt="Aura Chiikawa"
          class="_targetImage_c5ad1c false"
          src="/src/assets/temp/chiikawa.jpg"
        />
        <h4
          class=""
        >
          Aura Chiikawa
        </h4>
      </li>
    </ul>
  </section>
</div>
`;
